server/app.js

```diff
@@ -1,38 +1,64 @@
-  );
-  res.header('Access-Control-Expose-Headers', 'Authorization')
-  next()
-})
-
-app.use('/', async (req, res) => {
-  const port = req.query.port
-  console.log(`Request received for port ${port}`)
-  const jsonResponse = await getJson(port)
-  res.json(jsonResponse)
-  console.log("Response sent!")
-})
-
-app.listen(listenningPort, () => {
-  console.log(`Assetto-corsa-server-backend is listening on port ${listenningPort}`)
-})
-
-const getJson = async(port) => {
-  const url = `http://localhost:${port}/JSON%7C`;
-
-  return await fetch(url)
-  .then(res => res.json())
-  .catch(err => {
-    console.error(err)
-    return { status: 404, error: "No server running at this address" }
+const bodyParser = require("body-parser");
+const filesRoutes = require('./routes/files.routes.js')
+const serverRoutes = require('./routes/server.routes')
+const userRoutes = require('./routes/user.routes')
+const whitelistRoutes = require('./routes/whitelist.routes')
+
+module.exports = () => {
+  const app = express()
+
+  // Parses the request body to json automatically
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({ extended: true }))
+
+  // Fixes the cors issues
+  app.use((req, res, next) => {
+    res.header('Access-Control-Allow-Origin', process.env.FRONTEND_URL)
+    res.header('Access-Control-Allow-Credentials', true)
+    res.header('Access-Control-Allow-Methods', 'DELETE, PUT, POST, GET, PATCH')
+    res.header('Access-Control-Allow-Headers', 'x-requested-with, Content-Type, Accept, Access-Control-Request-Method, Access-Control-Request-Headers, Authorization, refreshtoken');
+    res.header('Access-Control-Expose-Headers', 'Authorization')
+    next()
   })
-}
+
+  // Routes
+  app.use('/files', filesRoutes)
+  app.use('/server', serverRoutes)
+  app.use('/user', userRoutes)
+  app.use('/whitelist', whitelistRoutes)
+
+  // TEMPORARY ADD to support old version while tests are made
+  app.use('/', async (req, res) => {
+    const port = req.query.port
+    console.log(`Request received for port ${port}`)
+
+    const url = `http://localhost:${port}/JSON%7C`;
+  
+    const jsonResponse = await fetch(url)
+    .then(res => res.json())
+    .catch(err => {
+      console.error(err)
+      return { status: 404, error: "No server running at this address" }
+    })
+
+    res.json(jsonResponse)
+    console.log("Response sent!")
+  })
+
+  // // Needs to be un commented when migration is done
+  // // Handles requests on base url
+  // app.use('/', async (req, res) => 
+  //   res.json({ message: "Hello from the Assetto Corssa Server Dashboard!" })
+  // )
+
+  // ****** To be added later on (This is just a template/idea) *******
+  // Handles page refresh on the client side
+  // (view index.hmtl and 404.html located in the client/public folder)
+  // app.use((req, res) =>
+    // res.sendFile(path.resolve(__dirname, '../client/public/404.html'))
+  // )
```

Comment:
Don't forget to remove the TEMPORARY code block and uncomment the commented code here

Reply:
No need to do this since when the backend will be deployed on the server, the front end will follow anyways. So, no need to keep support for the old front end.
Fixed in commit [4a6d2c4](https://github.com/Funnyadd/assetto-corsa-server-dashboard/pull/1/commits/4a6d2c4a39cc4feb487581a34796129df3372390)